### PR TITLE
Sync Main activity skeleton

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/0/1204195738264046/f 

### Description
Implement skeleton for sync main activity
Just UI elements
Will show toggle with expected state
If user signed in, show QR
Other parts of app are not functional

### Steps to test this PR

_Feature 1_ (to create or delete an account, use sync internal settings)
- [x] go to sync activity
- [x] if account exists toggle should be on
- [x] QR should be visible
- [x] if account doesn't exist, toggle should be off
- [x] account info should not be visible
- [x] hint is shown below toggle

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    private fun observeUiEvents() {
+        viewModel
+            .viewState()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.CREATED)
+            .onEach { viewState -> renderViewState(viewState) }
+            .launchIn(lifecycleScope)
+
+        viewModel
+            .commands()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
```
Comment (malmstein) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt:79]: nit: `Lifecycle.State.CREATED`

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+        setupToolbar(binding.includeToolbar.toolbar)
+        observeUiEvents()
+        configureListeners()
+    }
+
+    private fun configureListeners() {
+        binding.deviceSyncStatusToggle.setOnCheckedChangeListener(deviceSyncStatusToggleListener)
```
Comment (malmstein) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt:67]: you don't need this if you are calling `quietlySetIsChecked` afterwards

File: sync/sync-impl/src/main/res/drawable/ic_device_desktop_24.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:pathData="M4.875,3C3.287,3 2,4.287 2,5.875V14.125C2,15.713 3.287,17 4.875,17H10V19H7C6.448,19 6,19.448 6,20C6,20.552 6.448,21 7,21H17.125C17.677,21 18.125,20.552 18.125,20C18.125,19.448 17.677,19 17.125,19H14V17H19.125C20.713,17 22,15.713 22,14.125V5.875C22,4.287 20.713,3 19.125,3H4.875ZM4,5.875C4,5.392 4.392,5 4.875,5H19.125C19.608,5 20,5.392 20,5.875V13.125C20,13.608 19.608,14 19.125,14H4.875C4.392,14 4,13.608 4,13.125V5.875Z"
+      android:fillColor="#000000"
+      android:fillAlpha="0.84"
```
Comment (malmstein) [inline sync/sync-impl/src/main/res/drawable/ic_device_desktop_24.xml:9]: icons should not have alpha, `android:fillColor="?attr/daxColorPrimaryIcon"` instead.

File: sync/sync-impl/src/main/res/drawable/ic_device_mobile_24.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:pathData="M10.5,5C10.224,5 10,5.224 10,5.5C10,5.776 10.224,6 10.5,6H13.5C13.776,6 14,5.776 14,5.5C14,5.224 13.776,5 13.5,5H10.5Z"
+      android:fillColor="#000000"
```
Comment (malmstein) [inline sync/sync-impl/src/main/res/drawable/ic_device_mobile_24.xml:8]: ditto, remove `fillAlpha`, use proper `fillColor`

File: sync/sync-impl/src/main/res/drawable/ic_device_mobile_24.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+      android:fillAlpha="0.84"/>
+  <path
+      android:pathData="M7.875,2C6.287,2 5,3.287 5,4.875V19.125C5,20.713 6.287,22 7.875,22H16.125C17.713,22 19,20.713 19,19.125V4.875C19,3.287 17.713,2 16.125,2H7.875ZM7,4.875C7,4.392 7.392,4 7.875,4H16.125C16.608,4 17,4.392 17,4.875V19.125C17,19.608 16.608,20 16.125,20H7.875C7.392,20 7,19.608 7,19.125V4.875Z"
+      android:fillColor="#000000"
```
Comment (malmstein) [inline sync/sync-impl/src/main/res/drawable/ic_device_mobile_24.xml:12]: ditto, remove `fillAlpha`, use proper `fillColor`

File: sync/sync-impl/src/main/res/drawable/ic_device_tablet_24.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:pathData="M22,6.875C22,5.287 20.713,4 19.125,4H4.875C3.287,4 2,5.287 2,6.875V17.125C2,18.713 3.287,20 4.875,20H19.125C20.713,20 22,18.713 22,17.125V6.875ZM19.125,6C19.608,6 20,6.392 20,6.875V17.125C20,17.608 19.608,18 19.125,18L4.875,18C4.392,18 4,17.608 4,17.125L4,6.875C4,6.392 4.392,6 4.875,6L19.125,6ZM19,11.5C19,11.224 18.776,11 18.5,11C18.224,11 18,11.224 18,11.5V12.5C18,12.776 18.224,13 18.5,13C18.776,13 19,12.776 19,12.5V11.5Z"
+      android:fillColor="#000000"
```
Comment (malmstein) [inline sync/sync-impl/src/main/res/drawable/ic_device_tablet_24.xml:8]: ditto, use proper `fillColor`

**Review by malmstein — APPROVED** [review]
the vector drawables have the wrong fillColor, reminds me to add a lint rule for it. other than that it's all looking good.

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt (hunk 1 above)
Comment (cmonfortep) [reply sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt:79]: good catch!

File: sync/sync-impl/src/main/res/drawable/ic_device_desktop_24.xml (hunk 3 above)
Comment (cmonfortep) [reply sync/sync-impl/src/main/res/drawable/ic_device_desktop_24.xml:9]: thanks for also saying which color we should use 👍

**Pull request merged**